src/main/java/io/gravitee/connector/http/Http2Connector.java

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.connector.http;
+
+import io.gravitee.connector.api.EndpointException;
+import io.gravitee.connector.http.endpoint.HttpEndpoint;
+import io.gravitee.gateway.api.proxy.ProxyRequest;
+import io.gravitee.node.api.configuration.Configuration;
+import io.netty.handler.codec.http.HttpHeaderNames;
+import io.vertx.core.http.HttpClientOptions;
+import io.vertx.core.http.HttpVersion;
+import java.net.URL;
+
+/**
+ * @author GraviteeSource Team
+ */
+public class Http2Connector extends AbstractHttpConnector<HttpEndpoint> {
+
+    public Http2Connector(HttpEndpoint endpoint, Configuration configuration) {
+        super(endpoint, configuration);
+    }
+
+    @Override
+    public HttpClientOptions createHttpClientOptions() throws EndpointException {
+        HttpClientOptions options = super.createHttpClientOptions();
+
+        // Force HTTP/2 protocol
+        options.setProtocolVersion(HttpVersion.HTTP_2);
+
+        return options;
+    }
+
+    @Override
+    protected void translateHeaders(URL url, ProxyRequest request, String host) {
+        // TODO: support HTTP2 headers
+        //        request.headers().set(":authority", host);
+        //        if (!request.headers().contains(":scheme")) {
+        //            request.headers().set(":scheme", request.headers().get("X-Forwarded-Proto"));
+        //        }
+        //        if (!request.headers().contains(":path")) {
+        //            request.headers().set(":path", url.getFile());
+        //        }
+        // Strip all headers made illegal in HTTP/2 messages:
+        request.headers().remove(HttpHeaderNames.HOST);
```

Comment:
Could have a hop-by-hop list specific for HTTP2 as we already have for HTTP1.0/1.1 ?

Reply:
✅